engine/src/main/java/com/sandboni/core/engine/finder/bcel/visitors/CallerClassVisitor.java

```diff
@@ -36,14 +38,39 @@ public void visitMethod(Method method) {
 
         // in order to detect locations (modules) each method has to have at least one entry in the graph as caller
         if (linksAdded == 0) {
-            Vertex v ;
-            if (Objects.nonNull(context.getCurrentLocation()) && context.getTestLocations().contains(context.getCurrentLocation())){
+            Vertex v;
+            if (isTestLocation()) {
                 v = new TestVertex.Builder(this.javaClass.getClassName(), formatMethod(method), context.getCurrentLocation()).build();
-            }else{
+            } else {
                 v = new Vertex.Builder(this.javaClass.getClassName(), formatMethod(method), context.getCurrentLocation())
                         .build();
             }
-            context.addLink(LinkFactory.createInstance(context.getApplicationId(), v , DEAD_END_VERTEX, LinkType.METHOD_CALL));
+            context.addLink(LinkFactory.createInstance(context.getApplicationId(), v, DEAD_END_VERTEX, LinkType.METHOD_CALL));
         }
     }
+
+    private boolean isTestLocation() {
+        return Objects.nonNull(context.getCurrentLocation()) && context.getTestLocations().contains(context.getCurrentLocation());
+    }
+
+    @Override
+    public void visitConstantPool(final ConstantPool obj) {
```

Comment:
Did you measure the performance impact of this operation? seems pretty expensive. Let's confirm before merging this.

Reply:
From where is this called?

Reply:
it's called from com/sandboni/core/engine/finder/bcel/visitors/ClassVisitorBase.java:25
The constant pool can only give me method calls and object types for the invocation.
I cannot deduct what is the method from which this method call is done.
Hence, I need to create a link with the class name only, and include all tests in this class later on.